admin_manual/configuration_user/user_auth_ldap.rst

```diff
@@ -466,6 +467,84 @@ In migrated Nextcloud installations the old behavior still applies, which is usi
 
   sudo -u www-data php occ config:app:set user_ldap enforce_home_folder_naming_rule --value=1
 
+"$home" Placeholder Field:
+  $home in an external storage configuration will be replaced with the value of the specified attribute
+
+.. _LDAP_User_Profile_Attributes:
+
+User Profile attributes
+^^^^^^^^^^^^^^^^^^^^^^^
+
+.. figure:: ../images/ldap-advanced-4-attributes.png
+   :alt: User Profile Attributes.
+
+After configuring those attributes, the User Profile data will be overwritten with the according data from LDAP.
+The data must be formatted like Nextcloud wants it, otherwise it will be constantly changed and rechanged/removed again.
+
+:Please be aware:
+- The user can not change the data in profile, for it will get overwritten almost instantly
+- The user can change the visibility scope in profile
+- The default visibility can be adjusted with setting the ``account_manager.default_property_scope`` array in ``config.php``
+- If multiple attribute values are present, only the first value is used
+- All user profile properties are limited to 2048 character
+- Having misformatted data in LDAP will most probably leave you with empty user profile fields
+- Setting the global ``profile.enabled => false`` on ``config.php`` skips the code
+
+By calling ``php occ ldap:check-user --update <uid>`` the users data from LDAP will be displayed and the profile gets updated.
+
+.. note:: After unsetting or removing an profile attribute, the data wont be reset or deleted from user profile.
```

Comment:
No. If attribute name is not present in configuration or attribute is not present in the LDAP result, the user profile is ignored.
So, the users without admin defined values in LDAP can modify the profile and set what they like.
Otherwise the LDAP data should be written to user settings and user profile only updated/removed, after LDAP data changes. At least in my opinion.

Reply:
> No. If attribute name is not present in configuration or attribute is not present in the LDAP result, the user profile is ignored. So, the users without admin defined values in LDAP can modify the profile and set what they like. Otherwise the LDAP data should be written to user settings and user profile only updated/removed, after LDAP data changes. At least in my opinion.

For me this is not the same thing.
When the attribute name is empty in configuration yes, it should be left alone.

But if the attribute name is set and its value is empty(unset) in LDAP, it should empty the value in Nextcloud.

Otherwise, this happens:

1. Admin maps phone number to field phoneNumber
2. User Bob gets a phone, admin sets phoneNumber in LDAP to the phone number
3. Nextcloud sync that to phone number field
4. Bob loses phone, admin removes phone number from LDAP
5. Nextcloud ignores this changes and let the outdated data in the profile :warning:

Reply:
Okay. I see your point @come-nc and I'll fix it this weekend.
The fix could be populating the array with empty values. I had this and removed it for reasons I can't remember.

Reply:
> I had this and removed it for reasons I can't remember.

1. User marc has his phone number manually set in profile
2. Admin configures in LDAP marc has no phone number